WhateverGreen/kern_igfx.cpp

```diff
@@ -1058,47 +1058,55 @@ bool IGFX::applySklAsKblPatches(IOService *that) {
 	DBGLOG("igfx", "disabling VP9 hw decode support on Skylake with KBL kexts");
 	that->removeProperty("IOGVAXDecode");
 
-	auto hevcDecodeCap = OSDynamicCast(OSDictionary, that->getProperty("IOGVAHEVCDecodeCapabilities"));
-	if (!hevcDecodeCap)
-		return false;
+	// SKL does not support 10-bit hardware encoding/decoding, and this causes freezing when attempting to do so.
+	// The removal of profile 2 under VTSupportedProfileArray allows fallback to software encoding/decoding.
+	// Thanks dhinakg and aben for finding this.
+	const char *hevcCapProps[] = { "IOGVAHEVCDecodeCapabilities", "IOGVAHEVCEncodeCapabilities" };
+	for (auto prop : hevcCapProps) {
+		auto hevcCap = OSDynamicCast(OSDictionary, that->getProperty(prop));
+		if (!hevcCap)
+			return false;
```

Comment:
Why false? Maybe continue here and below?

Reply:
Well, not having decoder or encoder feels ok to me. Apple may update the profiles at their preference.

Reply:
https://github.com/acidanthera/WhateverGreen/commit/02ab31c6263a80571fa18b8c914df03585ce1846